app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java

```diff
@@ -238,19 +191,29 @@ public void onClick(final DialogInterface dialog, final int which) {
 
     public class GrpcTask extends AsyncTask<AbstractGrpcTaskExecutor, Void, String> {
 
+        private Context mContext;
+
+        public GrpcTask(final Context context) {
+            this.mContext = context;
+        }
+
         @Override
-        protected void onPreExecute() {
+        protected final void onPreExecute() {
             super.onPreExecute();
             enableButtons(Boolean.FALSE);
         }
 
         @Override
-        protected String doInBackground(final AbstractGrpcTaskExecutor... params) {
+        protected final String doInBackground(final AbstractGrpcTaskExecutor... params) {
             try {
+                final SharedPreferencesHandler sharedPreferences =
+                        new SharedPreferencesHandler(mContext);
```

Comment:
Yea, I can use instead of mContext MainActivity.this. But it is temporary code and will be moved in separate class. And I will need Context for getting SharedPreferences later.